x/params/types/subspace.go

```diff
@@ -84,13 +84,16 @@ func (s Subspace) transientStore(ctx sdk.Context) sdk.KVStore {
 
 // Validate attempts to validate a parameter value by its key. If the key is not
 // registered or if the validation of the value fails, an error is returned.
-func (s Subspace) Validate(ctx sdk.Context, key []byte, value interface{}) error {
+func (s Subspace) Validate(ctx sdk.Context, key []byte, newValue interface{}) error {
 	attr, ok := s.table.m[string(key)]
 	if !ok {
 		return fmt.Errorf("parameter %s not registered", string(key))
 	}
 
-	if err := attr.vfn(value); err != nil {
+	currentValue := reflect.Indirect(reflect.ValueOf(newValue)).Interface()
```

Comment:
not sure if this is the right way to do this. @alessio can you take a look? 🙏

Reply:
It kind of smells correct. There is no unit testing for Validate() currently. How about writing one?